File: app/src/main/java/com/futnote/mobile/room/NoteDao.java
package com.futnote.mobile.room;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.futnote.mobile.models.Note;

import java.util.List;

@Dao
public interface NoteDao {

    @Insert
    void insert(Note note);

    @Update
    void update(Note note);

    @Delete
    void delete(Note note);

    @Query("DELETE FROM tblnote")
    void deleteAll();

    @Query("SELECT * FROM tblNote ORDER BY id DESC")
    LiveData<List<Note>> select();
}

File: app/src/main/java/com/futnote/mobile/room/NoteDao.kt
package com.futnote.mobile.room

import androidx.lifecycle.LiveData
import androidx.room.*
import com.futnote.mobile.model.Note

@Dao
interface NoteDao {
    @Insert
    fun insert(note: Note?)

    @Update
    fun update(note: Note?)

    @Delete
    fun delete(note: Note?)

    @Query("SELECT * FROM tblNote ORDER BY id DESC")
    fun select(): LiveData<List<Note?>?>?
}